view/frontend/templates/button_product_page.phtml

```diff
@@ -164,52 +163,45 @@ if (!$block->isSupportableType()) return;
                 }
             };
 
-            function setupProductPage() {
-                var quantity = 1;
-                if ($qty) {
-                    quantity = Number($qty.val());
-                }
+            var getQty = function() {
+                var quantity = Number($('#qty').val());
+                return quantity > 0 ? quantity : 1;
+            };
+
+            var setupProductPage = function () {
                 var cart = {
                     currency: "USD",
                     items: [{
                         reference: '<?= $block->getProduct()->getId() ?>',
                         price: <?= (int) $block->getProduct()->getFinalPrice() ?>,
                         name: '<?= $block->getProduct()->getName() ?>',
-                        quantity: quantity,
+                        quantity: getQty(),
                     }]
                 };
 
                 var buttonClass = "bolt-product-checkout-button";
-                var hints = {prefill:{}};
-
-                // We need to receive hints from server
-                // meanwhile initiate Bolt without hints to have button early
-                BoltCheckout.configureProductCheckout(cart, hints, callbacks, {checkoutButtonClassName: buttonClass});
-                if ( customer().firstname ) {
```

Comment:
I wanted to fetch hints from the server only for logged in users. I think we should check it looking from a performance perspective.
But I agree we can grab hints for guest users also for now and change the behavior in the next version.